src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,111 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace; ?>;
+
+use <?= $token_full_class_name ?>;
+use <?= $user_full_class_name ?>;
+use <?= $request_form_full_class_name ?>;
+use <?= $resetting_form_full_class_name ?>;
+use Symfony\Bundle\FrameworkBundle\Controller\<?= $parent_class_name; ?>;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+
+class <?= $class_name; ?> extends <?= $parent_class_name; ?><?= "\n" ?>
+{
+    /**
+     * @Route("/forgotten-password", name="app_forgotten_password")
+     */
+    public function request(Request $request, \Swift_Mailer $mailer): Response
+    {
+        $form = $this->createForm(<?= $request_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+                '<?= $email_field ?>' => $form->get('<?= $email_field ?>')->getData(),
+            ]);
+
+            // do not reveal whether an user account was found or not
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, that the user could use to change his password
+            $resetPasswordToken = new <?= $token_class_name?>($user);
+            $this->getDoctrine()->getManager()->persist($resetPasswordToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            $message = (new \Swift_Message('Your password reset request'))
+                ->setFrom(['noreply@mydomain.com' => 'Noreply'])
+                ->setTo($user-><?= $email_getter ?>())
+                ->setBody($this->renderView('forgottenPassword/email.txt.twig', [
+                    'token' => $resetPasswordToken,
+                ]))
+            ;
+            $mailer->send($message);
+
+            return $this->redirectToRoute('app_check_email');
+        }
+
+        return $this->render('forgottenPassword/request.html.twig', [
+            'requestForm' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/check-email", name="app_check_email")
+     */
+    public function checkEmail()
+    {
+        return $this->render('forgottenPassword/check_email.html.twig', [
+            'tokenLifetime' => <?= $token_class_name ?>::TTL_HOURS,
+        ]);
+    }
+
+    /**
+     * @Route("/reset-password/{tokenAndSelector}", name="app_reset_password")
```

Comment:
And the last but not least important feature I would like to see in this PR - store token in session and hide it from the URL. It means we should make `tokenAndSelector` optional, and if the `tokenAndSelector` is passed - we store it in session and redirect users to just `/reset-password/`, but if there's no token in the URL - we read it from the session. If there's no token in both places - `createNotFoundException()` then. Wdyt?

This is a pretty important feature btw, because some JS scripts may reveal token to other third-party services when sending AJAX requests from this page, so this is important for security reasons.

Reply:
I just modified it. And I forgot to react to your comment, but actually, that's a nice catch and a smart feature 👍